Ilya_prokofev/Source.cpp

```diff
@@ -0,0 +1,377 @@
+#include <string>
+#include <iostream>
+#include <fstream>
+#include <vector>
+#include <set>
+
+using namespace std;
+
+template < typename K, typename V>
+struct node {
+	K key;
+	V value;
+	node* next;
+	node(K k, V v, node* next) : key(k), value(v), next(next) {}
+};
+template < typename K, typename V>
+struct list {
+	node<K,V>* head;
+	list() : head(nullptr) {}
+	void push(K k, V v) {
+		node<K, V>* new_node = new node <K, V>(k, v, head);
+		head = new_node;
+	}
+	size_t size() {
+		size_t cnt = 0;
+		node<K, V>* tmp = head;
+		while (tmp) {
+			cnt++;
+			tmp = tmp->next;
+		}
+		return cnt;
+	}
+	vector<pair<K, V>> get_pairs() {
+		node<K, V>* tmp = head;
+		vector<pair<K, V>> res;
+		while (tmp) {
+			res.push_back(make_pair(tmp->key, tmp->value));
+			tmp = tmp->next;
+		}
+		return res;
+	}
+	void rewrite(K k, V v) {
+		node<K, V>* tmp = head;
+		while (tmp != nullptr) {
+			if (tmp->key == k) {
+				tmp->value = v;
+				return;
+			}
+			tmp = tmp->next;
+		}
+		push(k, v);
+	}
+	void remove(K k) {
+		node <K, V>* prev = nullptr;
+		node<K, V>* current = head;
+		while (current) {
+			if (current->key == k) {
+				if (prev) {
+					prev->next = current->next;
+				}
+				else {
+					head = current->next;
+				}
+				delete current;
+				return;
+			}
+			prev = current;
+			current = current->next;
```

Comment:
JFYI: https://medium.com/@bartobri/applying-the-linus-tarvolds-good-taste-coding-requirement-99749f37684a